myia/abstract/utils.py

```diff
@@ -694,11 +702,11 @@ def chk(a):
             f'Type mismatch: {type(x1)} != {type(x2)}; {x1} != {x2}'
         )
     else:
-        return __call__(x1, x2, loop, forced, bind_pending)
+        return self.map[type(x1)](self, x1, x2, forced, bind_pending)
```

Comment:
Why are you using an overload wrapper if you just ignore the result of the overload resolution?

Reply:
Because of the line that goes `x1 = x1.result()` if `x1` is `Pending`. Without this adjustment, it would call the function for `Pending`. I suppose I could replace that line with a return of a recursive call, though.